packages/picasso.js/src/core/component/component-factory.js

```diff
@@ -449,11 +449,12 @@ function componentFactory(definition, context = {}) {
 
   function updateBrushNodes(nodes) {
     const { rendererSettings } = settings;
-    const progressive = typeof rendererSettings?.progressive === 'function' && rendererSettings.progressive();
-    if (progressive && brushArgs.nodes) {
-      brushArgs.nodes.push(...nodes);
-    } else {
+    if (typeof rendererSettings?.progressive !== 'function') {
       brushArgs.nodes = nodes;
+    } else if (rendererSettings.progressive() && brushArgs.nodes) {
+      brushArgs.nodes.push(...(nodes || []));
+    } else {
+      brushArgs.nodes = [...(nodes || [])];
     }
```

Comment:
why not
```
    const progressive = typeof rendererSettings?.progressive === 'function' && rendererSettings.progressive();
    if (progressive && brushArgs.nodes) {
      brushArgs.nodes = [...brushArgs.nodes, ...nodes];
    } else {
      brushArgs.nodes = nodes;
    }
```
(maybe with an extra `|| []` if needed)
I think it is safer and more understandable

Reply:
I want to distinguish between progressive undefined and progressive false.